jujugui/static/gui/src/app/assets/javascripts/jujulib/test-juju.js

```diff
@@ -5,14 +5,86 @@
 chai.config.includeStack = true;
 chai.config.truncateThreshold = 0;
 
-describe('juju.js', function() {
-
-    it('exists', function() {
-        var cs = new jujulib.charmstore();
-        var env = new jujulib.environment();
-        var identity = new jujulib.identity();
-        assert.isTrue(cs instanceof jujulib.charmstore);
-        assert.isTrue(env instanceof jujulib.environment);
-        assert.isTrue(identity instanceof jujulib.identity);
+describe('jujulib', function() {
+
+    describe('environment manager', function() {
+        it('exists', function() {
+            var env = new window.jujulib.environment();
+            assert.isTrue(env instanceof window.jujulib.environment);
+        });
+
+        it('can list environments', function(done) {
+            var bakery = {
+                sendGetRequest: function(path, ignore, success, failure) {
+                    success({target: {responseText: '{"environments": ["foo"]}'}});
```

Comment:
Perhaps a makeXhrResponse function would help here, which takes an object and returns `{target: {responseText: jsonOf(obj)}}`